app/src/main/java/com/yalta/activities/ChangePasswordActivity.kt

```diff
@@ -0,0 +1,81 @@
+package com.yalta.activities
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.text.Selection
+import android.view.View
+import android.view.animation.DecelerateInterpolator
+import android.view.inputmethod.EditorInfo
+import androidx.databinding.DataBindingUtil
+import androidx.lifecycle.ViewModelProvider
+import com.yalta.R
+import com.yalta.databinding.ActivityChangePasswordBinding
+import com.yalta.viewmodel.ChangePasswordViewModel
+import kotlinx.android.synthetic.main.activity_change_password.*
+
+class ChangePasswordActivity : AppCompatActivity() {
+    private val viewModel by lazy { ViewModelProvider(this).get(ChangePasswordViewModel::class.java) }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        overridePendingTransition(0, 0)
+
+        val binding: ActivityChangePasswordBinding =
+            DataBindingUtil.setContentView(this, R.layout.activity_change_password)
+
+        binding.viewModel = viewModel
+        binding.lifecycleOwner = this
+
+        binding.viewModel?.closeActivity?.observeForever { closeActivity ->
+            if (closeActivity) {
+                close_button.performClick()
+            }
+        }
+
+        binding.viewModel?.showError?.observeForever { showError ->
+            error.visibility = if (showError) View.VISIBLE else View.INVISIBLE
```

Comment:
Hmm, I would guess that after WPF visibility can be bind to bool (Al least with converter) please bind it, and not do this

Reply:
There are three states: INVISIBLE, VISIBLE and GONE, so no :/

Reply:
Lul, they never learn...

Reply:
Updated